graphjet-core/src/main/java/com/twitter/graphjet/algorithms/counting/user/TopSecondDegreeByCountForUser.java

```diff
@@ -48,21 +49,26 @@ public TopSecondDegreeByCountForUser(
 
   @Override
   protected void updateNodeInfo(
+    TopSecondDegreeByCountRequestForUser request,
     long leftNode,
     long rightNode,
     byte edgeType,
     double weight,
-    EdgeIterator edgeIterator,
-    int maxSocialProofTypeSize) {
+    EdgeIterator edgeIterator) {
     NodeInfo nodeInfo;
 
+    long segmentCreationTime = ((TimestampEdgeIterator)edgeIterator).currentSegmentCreationTime();
+    if (segmentCreationTime < request.getEarliestValidTimeForSocialProof()) {
```

Comment:
in RecentTweetFilter.java, there is similar functionality. clients pass a variable called keepWithInLastXMillis, and the tweet creation time is compared against (System.currentTimeMillis() - keepWithInLastXMillis). in that implementation, the client only needs to concern about the keepWithInLastXMillis, and it could be 24 * 60 * 60 * 1000. In your current implementation, getEarliestValidTimeForSocialProof changes over time.